exporters/ostream/include/opentelemetry/exporters/ostream/span_exporter.h

```diff
@@ -86,62 +91,7 @@ class OStreamSpanExporter final : public sdktrace::SpanExporter
 
   void print_value(sdktrace::SpanDataAttributeValue &value)
   {
-    if (nostd::holds_alternative<bool>(value))
-    {
-      sout_ << nostd::get<bool>(value);
-    }
-    else if (nostd::holds_alternative<int32_t>(value))
-    {
-      sout_ << nostd::get<int32_t>(value);
-    }
-    else if (nostd::holds_alternative<uint32_t>(value))
-    {
-      sout_ << nostd::get<uint32_t>(value);
-    }
-    else if (nostd::holds_alternative<int64_t>(value))
-    {
-      sout_ << nostd::get<int64_t>(value);
-    }
-    else if (nostd::holds_alternative<uint64_t>(value))
-    {
-      sout_ << nostd::get<uint64_t>(value);
-    }
-    else if (nostd::holds_alternative<double>(value))
-    {
-      sout_ << nostd::get<double>(value);
-    }
-    else if (nostd::holds_alternative<std::string>(value))
-    {
-      sout_ << nostd::get<std::string>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<bool>>(value))
-    {
-      print_array<bool>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<int32_t>>(value))
-    {
-      print_array<int32_t>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<uint32_t>>(value))
-    {
-      print_array<uint32_t>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<int64_t>>(value))
-    {
-      print_array<int64_t>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<uint64_t>>(value))
-    {
-      print_array<uint64_t>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<double>>(value))
-    {
-      print_array<double>(value);
-    }
-    else if (nostd::holds_alternative<std::vector<std::string>>(value))
-    {
-      print_array<std::string>(value);
-    }
+    nostd::visit([this](auto &&arg) { print_value(arg); }, value);
```

Comment:
It seems missing templatized lambdas in C++11 is limiting this simplification.

Reply:
I wonder if we could pass the overloaded `print_value` function to `nostd::visit` right away, without wrapping it in a lambda.

Reply:
For passing `print_value` function to `nostd::visit`, we still need a way to pass `this` or `std::ostream` to the type specific `print_value`. Perhaps I could add a functor to do this.

Reply:
[std::bind](https://en.cppreference.com/w/cpp/utility/functional/bind) can help here:
```c++
// bind to a pointer to member function
Foo foo;
auto f3 = std::bind(&Foo::print_sum, &foo, 95, _1);
f3(5);
```

Reply:
`std::bind` may not help here because `nostd::visit()` requires visitor to be exhaustive on the types of visiting target.